nimbleclient/v1/resource.py

```diff
@@ -3,21 +3,32 @@
 #
 
 class Resource:
-    __slots__ = ['id', 'attrs', 'collection', '_client']
+    __slots__ = ['_id', '_attrs', '_collection', '_client']
```

Comment:
This defeats the purpose. `.id` and `.attrs` should be accessible by the user.

Reply:
Any reason why this should be exposed to the users directly? It's better to have accessor methods for these so that no one can modify them inadvertently.

Reply:
> This defeats the purpose. `.id` and `.attrs` should be accessible by the user.

It will still be accessible by the user in the same way but readonly (cannot set/change these values to something else). 
For ex: volume.id = "something"  ==>> This is not allowed

Reply:
The general Python philosophy is to give users the freedom to hang themselves where possible. If you really want to lock it down though, you have to use double_underscore: `__id` not `_id`

Reply:
Agree. Python makes attributes hidden when used with __ (double underbar), although technically they are still accessible via __dict__ interface, but accidental misuse via that route is very rare.